packages/java/src/node/java-service.ts

```diff
@@ -0,0 +1,67 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { NodePackage, ExtensionPackage } from '@theia/application-package';
+import { JavaService } from '@theia/java/src/common/java-common';
+import { ApplicationPackageManager } from '@theia/application-manager';
+import { injectable } from 'inversify';
+import * as paths from 'path';
+
+interface JavaContributes extends NodePackage {
+    javaExtensions?: string[];
+}
+
+/**
+ * JavaService implementation.
+ */
+@injectable()
+export class JavaServiceImpl implements JavaService {
+    private javaExtensions: string[] | undefined;
+
+    async collectJavaExtensions(): Promise<string[] | undefined> {
+        if (this.javaExtensions) {
+            return Promise.resolve(this.javaExtensions);
+        }
+
+        const projectPath = process.cwd();
+        const manager = new ApplicationPackageManager({ projectPath });
+        const extensions = manager.pck.extensionPackages;
+        extensions.forEach(extensionInfo => {
+            this.findBundles(extensionInfo);
+        });
+        return this.javaExtensions;
+    }
+
+    private findBundles(extPck: ExtensionPackage) {
+        if (!this.javaExtensions) {
+            this.javaExtensions = [];
+        }
+        const contributes: JavaContributes = extPck.raw.contributes;
```

Comment:
Sorry, but I don't like this part at all. We should not be nosing around in other extensions package.json. We should be defining an extension interface and let the contributing extensions decide how the want to give us a list of extension jars. Like 
```
interface JdtLsExtension {
    getExtensionJars() : String
}
```
We can still use package.json for a default implementation that contributors can use.

Reply:
I still don't understand why declaration in package.json is bad idea. As for me it is simpler to add a path to the my bundle into packaje.json instead of implement some interface and it will work for both: extensions and plugins in Theia.